python/felis/validation.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of felis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+from typing import Any, Sequence, Type
+
+from pydantic import Field, model_validator
+
+from .datamodel import Column, Schema, Table
+
+logger = logging.getLogger(__name__)
+
+
+class RspColumn(Column):
+    """Column for RSP data validation."""
+
+    description: str = Field(..., min_length=1)
+    """Redefine description to make it required and non-empty.
+    """
+
+
+class RspTable(Table):
+    """Table for RSP data validation.
+
+    The list of columns is overridden to use RspColumn instead of Column.
+
+    Tables for the RSP must have a TAP table index and a valid description.
+    """
+
+    description: str = Field(..., min_length=1)
```

Comment:
Can you make this test for non-whitespace?

Reply:
See https://github.com/lsst/felis/pull/31#discussion_r1488588606